include/hypercomm/aggregation.hpp

```diff
@@ -14,6 +14,11 @@
 #include <numeric>
 #include <tuple>
 
+// TODO(jszaday): this line should be corrected once a version
+//                of charm++ after v7.0.0 is released. it can't
+//                be correctly defined for now :(
+#define HYPERCOMM_CHARM_PAST_V_7_0_0 CHARM_VERSION_MAJOR >= 7
```

Comment:
Charm now defines a macro CCD_COND_FN_EXISTS that you could check.

Reply:
Ah, cool, that'd accomplish this goal 👍 While there's also the matter of `CkArray::deliver`, it was changed before `CCD_COND_FN_EXISTS` so it'd squeak through.